### PR TITLE
FLEX-4792 ~ Fixes error responses for light and tariff schedules.

### PR DESCRIPTION
**Pull request by kevinsmeets**

### CODE REVIEW

**Comment by kevinsmeets** [comment]
retest this please

File: integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-publiclighting/ScheduleManagement/SetLightSchedule.feature [hunk 1]
```diff
@@ -174,11 +171,77 @@ Feature: PublicLightingScheduleManagement Set Light Schedule
...
-      | ABSOLUTEDAY | ABSOLUTETIME | 18:00:00.000 |               | Validation Exception, violations: startDay may not be null when weekDay is set to ABSOLUTEDAY;               |
+      | WeekDay     | ActionTime   | Time         | TriggerType   | Message                                                                                                                                       |
+      | ABSOLUTEDAY | ABSOLUTETIME | 18:00:00.000 |               | Validation Exception, violations: startDay may not be null when weekDay is set to ABSOLUTEDAY;                                                |
       | MONDAY      | SUNRISE      |              | LIGHT_TRIGGER | Validation Exception, violations: triggerWindow may not be null when actionTime is set to SUNRISE or SUNSET and triggerType is LIGHT_TRIGGER; |
       | MONDAY      | SUNSET       |              | LIGHT_TRIGGER | Validation Exception, violations: triggerWindow may not be null when actionTime is set to SUNRISE or SUNSET and triggerType is LIGHT_TRIGGER; |
 
+  Scenario Outline: Set light schedule with for inactive/unregistered device (device lifecycle state)
```
Comment (rlemmers) [inline integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-publiclighting/ScheduleManagement/SetLightSchedule.feature]: "with for" => "for"
Also change this for all the other new scenarios.

File: osgp/platform/osgp-adapter-ws-publiclighting/src/main/java/org/opensmartgridplatform/adapter/ws/publiclighting/infra/jms/messageprocessor/DomainResponseMessageProcessor.java [hunk 2]
```diff
@@ -125,10 +125,11 @@ protected void handleMessage(final CorrelationIds ids, final String messageType,
 
         final short numberOfNotificationsSent = 0;
         Serializable deviceResponseObject;
-        if (dataObject == null) {
-            deviceResponseObject = resultDescription;
-        } else {
+
+        if (ResponseMessageResultType.OK.equals(resultType)) {
```
Comment (rlemmers) [inline osgp/platform/osgp-adapter-ws-publiclighting/src/main/java/org/opensmartgridplatform/adapter/ws/publiclighting/infra/jms/messageprocessor/DomainResponseMessageProcessor.java]: ResponseMessageResultType.OK == resultType is also allowed and is easier to read (in my opinion).
The same applies to the DomainResponseMessageProcessor for tariff switching.

File: integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-publiclighting/ScheduleManagement/SetLightSchedule.feature (hunk 1 above)
Comment (kevinsmeets) [reply integration-tests/cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-publiclighting/ScheduleManagement/SetLightSchedule.feature]: Done

File: osgp/platform/osgp-adapter-ws-publiclighting/src/main/java/org/opensmartgridplatform/adapter/ws/publiclighting/infra/jms/messageprocessor/DomainResponseMessageProcessor.java (hunk 2 above)
Comment (kevinsmeets) [reply osgp/platform/osgp-adapter-ws-publiclighting/src/main/java/org/opensmartgridplatform/adapter/ws/publiclighting/infra/jms/messageprocessor/DomainResponseMessageProcessor.java]: Done

**Pull request merged**